### ISSUE TITLE
ASP.NET Core 2.0 breaking changes

### ISSUE THREAD

**Issue by glennc** [Azure/service-fabric-aspnetcore#6]
This PR: https://github.com/aspnet/Hosting/pull/995

Removes the Start and Stop methods from IWebHost, replacing them with StartAsync and StopAsync. This will break the CommunicationListener here by the look of it.

Not a difficult change, but it is breaking for everyone, so you would presumably have to bump Major versions to depend on the new breaking Hosting package. How do you want to handle a change like this? In terms of branching and such for the most part.

**Comment by vturecek** [comment]
@glennc Yeah we've already run into this one. We don't currently have multiple major release support trains. Will asp.net core 2.0 still support full .NET framework or is it .NET Core only at that point?

**Comment by pksorensen** [comment]
I ran into this also, and the current codebase do work with 2.0 also, as there are extension methods added. Then building against the 2.0 feed instead:   `https://dotnet.myget.org/F/aspnetcore-ci-dev/api/v3/index.json` allowed me to use the kestrel communication listenere.

**Comment by Mardoxx** [comment]
Is [this](https://github.com/aspnet/Home/issues/2022) relevant to this discussion? I am not really following it, all the naming is really hard to get my head around.

By the sounds of it all future ASP.NET Core libraries will be targeting `netcoreapp2.0` which will _only_ run under .NET Core, and not .NET Framework. Reading a little more it appears that you will not be able to use .NET Framework libraries in a .NET Core application, and as far as I am aware SF libraries are .NET Framework only.

Is there going to be an issue here? This is so confusing. :/

**Comment by pksorensen** [comment]
I am would also like to know what the linked discussion actually means for service fabric. 

I have also been using dotnet core abstraction/extensions - interfaces and whatnot in my applications. And I am honestly not deep enough into what the discussion means to fully understand. Will my stuff start breaking?

Is it as simply as building the services for netcoreapp2.0 and target win10-x64 ? 

Would be nice if someone from the service fabric team or dotnet core team could recap what all this means.

**Comment by glennc** [comment]
For preview1 of ASP.NET Core 2.0 the ASP.NET packages had to drop support for .NET Framework. So you can only use netcoreapp2.0 apps. But that is just for the first preview while we get stuff to work. MVC and some other critical parts of the stack broke on full .net late in our development cycle.

You can consume .NET Framework libraries from a netcoreapp2.0 application, but there are no guarantees about whether or not it will work at runtime. If you happen to hit an API that doesn't work on Core then it will fail. You can fix that by recompiling the libraries to target netstandard at which point the library author will get warnings and errors about whether or not they will run. So targeting a netstandard version is the best way to go.

Does that make more sense now?

**Comment by vturecek** [comment]
Start/Stop have been added back to IWebHost for binary-compatibility. Now tracking behavioral change to support IPv6 in `IServerAddressesFeature` (from aspnet/Hosting#1131)
